### ISSUE TITLE
Always cast filter values into `int` considered harmful

### ISSUE THREAD

**Issue by vmeyet** [jhund/filterrific#34]
In the `ParamSet` class [when the value looks like an int we cast it as an int](https://github.com/jhund/filterrific/blob/59459b2f97cf73e0cf8a0f711b878376f1094f90/lib/filterrific/param_set.rb#L83-L85)

ex

```ruby
[1] > condition_filterrific_params { 'filter_array_int' => %w[1 2 3], 'filter_int' => '42' }
=> { 'filter_array_int' => [1, 2, 3], 'filter_int' => 42 }
```

which is fine most of the time.

however this might not always be what we want, for example, i want to look for a basic search on `:phone` in a `User` activerecord, I might do:

```ruby
class User < ActiveRecord::Base
    scope :with__phone__ilike, ->(value) { where("phone ILIKE ?", "%#{value}%") }
end

# let's say in db we have the following users
# User(phone: '12346789')
# User(phone: '12340067')

class SearchController < ActionController
    def search_phone
        filt = initialize_filterrific(User, params)
        users = User.filterrific_find(filt)
        render json: users.pluck(:phone)
    end
end

# calling the route with: params = { 'with__phone__ilike' => '067' }
# will sanitize the filter to { 'with__phone__ilike' => 67 }
# and returns false positive => ['12346789', '12340067']
```

*in that case the leading 0 was important in the filter value and was filter out because it was casted as int*.

*Same problem occurs if it appears in an array*

I think we might expect information not to be dropped (in that case the leading zero)
something like

```ruby
[1] > condition_filterrific_params { 'filter_array_int' => %w[1 2 3 03], 'filter_not_quite_int' => '042', 'filter_int' => '42' }
=> { 'filter_array_int' => [1, 2, 3, '03'], 'filter_not_quite_int' => '042', 'filter_int' => 42 }
```

this could be achieved by using a different regex such as `/^[1-9]\d*$/`. Replacing [that code](https://github.com/jhund/filterrific/blob/59459b2f97cf73e0cf8a0f711b878376f1094f90/lib/filterrific/param_set.rb#L74-L89) by

```ruby
def condition_filterrific_params(fp)
      fp.each do |key, val|
        case
        when val.is_a?(Proc)
          # evaulate Procs
          fp[key] = val.call
        when val.is_a?(Array)
          # type cast integers in the array
          fp[key] = fp[key].map { |e| e =~ /^[1-9]\d*$/ ? e.to_i : e }
        when val =~ /^[1-9]\d*$/
          # type cast integer
          fp[key] = fp[key].to_i
        end
      end
      fp
    end
``

**Comment by jhund** [comment]
@vmeyet thank you very much for making filterrific better! I just released a new version 2.0.4 that contains your fix.

**Comment by vmeyet** [comment]
Thanks, just saw the commit. It looks great! closing that Issue.
